### PR TITLE
Issue 390 task 2: Add switch chassis collection in the Chassis GET Collection response

### PR DESCRIPTION
**Pull request by a-ajith**
The PR contains:
- The code for fetching the chassis collection from all the fabric plugins available
- Unit test case for the same
- Fix for the go vet issues

This is for task 2 of #390

### CODE REVIEW

**Review by bharath-b-hpe — APPROVED** [review]

File: svc-systems/chassis/fabric.go [hunk 1]
```diff
@@ -0,0 +1,259 @@
...
+	PostBody        interface{}
+	LoginCredential map[string]string
+	Plugin          smodel.Plugin
+	Token           string
+}
+
+// PluginToken interface to hold the token
+type PluginToken struct {
+	Tokens map[string]string
+	lock   sync.Mutex
```
Comment (irappa-pattar) [inline svc-systems/chassis/fabric.go]: instead of `sync.Mutex` use `sync.RWMutex`

**Review by irappa-pattar — CHANGES_REQUESTED** [review]

**Review by skhan-anjum — APPROVED** [review]

**Review by Bharath-KKB — CHANGES_REQUESTED** [review]
Address Pattars code. This code is HUGE !! 8(

File: svc-systems/chassis/fabric.go [hunk 2]
```diff
@@ -0,0 +1,259 @@
...
+		return
+	}
+	for _, manager := range managers {
+		f.wg.Add(1)
+		go f.getFabricCollection(manager)
+	}
+	f.wg.Wait()
+}
+
+func (f *fabricFactory) getFabricCollection(plugin smodel.Plugin) {
```
Comment (jeevan-kamkar) [inline svc-systems/chassis/fabric.go]: rename the method  this method not collecting fabric information

**Review by jeevan-kamkar — CHANGES_REQUESTED** [review]
Address the changes and few code comments in the newly added function

File: svc-systems/chassis/fabric.go (hunk 1 above)
Comment (a-ajith) [reply svc-systems/chassis/fabric.go]: addressed

File: svc-systems/chassis/fabric.go (hunk 2 above)
Comment (a-ajith) [reply svc-systems/chassis/fabric.go]: addressed

**Review by jeevan-kamkar — APPROVED** [review]

**Review by irappa-pattar — APPROVED** [review]

File: svc-systems/chassis/fabric.go (hunk 1 above)
Comment (Bharath-KKB) [reply svc-systems/chassis/fabric.go]: Confirm we are using Lock()/UnLock() for writes and RLock()/RUnnLock() for read ops

Comment (a-ajith) [reply svc-systems/chassis/fabric.go]: @Bharath-KKB Checked

**Review by Bharath-KKB — APPROVED** [review]

**Pull request merged**